### ISSUE TITLE
OnNotificationOpened not being called when app is killed or removed from memory/force-quit

### ISSUE THREAD

**Issue by fernsheldon** [OneSignal/react-native-onesignal#70]
I went thru the documentation on GitHub, and understand that there is a small issue with iOS and OneSignal for ReactNative, where the notification is coming too fast. 

But here my issue is that i believe the callback `onNotification` is not being called when the app isn't in the background.

```
OneSignal.configure({
  onNotificationOpened: function(message, data, isActive) {
  var notification = {message: message, data: data, isActive: isActive};
  alert("going to handleNotification");
  handleNotification(notification);
  }
});
```

This is the snipped of code that is supposed to run as a callback for when a notification is opened. It runs perfectly fine when the app is opened and in the background.

But lets say I kill the app by double clicking the home button and removing the app. Then when a notification comes in and I click on it, I do not get the alert. This leads me to the conclusion that the call back is not being called. The app opens up fine, but that the payload in the notification isn't loaded because the method handleNotification isn't called.

ReactNative 0.31

OS: iOS 9.3

Any thoughts/ideas/solutions would be awesome.

**Comment by onchainguy-btc** [comment]
Could also be linked to #69

**Comment by jkasten2** [comment]
@avishayil I remember we had this issue a while back with #28 but I believe it was fixed. Could a recent change broken this?

**Comment by JKalash** [comment]
it may also be related to the fact that you're using 2.\* SDK on which is not yet supported (WiP, PR: https://github.com/geektimecoil/react-native-onesignal/pull/73) 

Temporary fix meanwhile: Modify your podfile: change 
**pod 'OneSignal'** to **pod 'OneSignal', '~> 1.13.2'**

**Comment by avishayil** [comment]
We're currently testing the new version, which apparently solves this issue.

**Comment by fernsheldon** [comment]
@avishayil that awesome !  Thanks

**Comment by fernsheldon** [comment]
@avishayil @JKalash So I tried the "1.13.3" version of the pod.  I don't think it helped out. 
Is there any work around, or an ETA for the new SDK ?

**Comment by fernsheldon** [comment]
Hi @avishayil @JKalash  !

So we found out a workaround  to get the notification data even when the app is forced closed by double clicking the home button and swiping the app away.  

This particular thread ,[8580](https://github.com/facebook/react-native/issues/8580) , proved to be super helpful. 

Here is the relevant part:

In the AppDelegate.m 

```
  // Inside of your didFinishLaunchingWithOptions method...
  // Create a Mutable Dictionary to hold the appProperties to pass to React Native.
    NSMutableDictionary *appProperties = [NSMutableDictionary dictionary];

    if (launchOptions != nil) {
      // Get Local Notification used to launch application.
      NSDictionary *notification = [launchOptions objectForKey:UIApplicationLaunchOptionsRemoteNotificationKey];

      if (notification) {
        [ appProperties setObject:notification forKey:@"initialNotification" ];
      }
    }

  .
  .
  .
  rootView.appProperties = appProperties;
```

Then in your index.ios.js: 

```
  componentDidMount() {

    if (this.props.initialNotification) {
      alert(JSON.stringify(this.props.initialNotification));
     }
  }
```

So while we were able to get the notification data, we had one _**other**_ issue.  Part of our app is loaded via a [WebView](http://facebook.github.io/react-native/releases/0.34/docs/webview.html) component.  Here we noticed that while the [props](http://facebook.github.io/react-native/releases/0.34/docs/webview.html#props) were working fine when the app is launched normally, they did not fire when the app was lunched via a PN after the app was forced killed.  

Initially we thought this was probably an error with RN itself.  But just to make sure we tested our app with PushWoosh's SDK.  To our amazement, WebView's props worked fine, regardless if the app was forced killed or launched normally.  This lead us to believe that the issue might be with OneSignal's SDK.

We just wanted to update this thread with all the information we had gathered so far. 

Hopefully this would get solved with your new SDK release 😃 .  

Do let us know when it's released.  
Waiting intently !
